### PR TITLE
Fix to umi_tools

### PR DESCRIPTION
**Pull request by mtekman**
* Added paired inputs for `extract`
 * Macrofied common options for `extract` and `whitelist`
 * Only `extract` wrapper version has been bumped due to added functionality.

### CODE REVIEW

**Comment by bgruening** [comment]
@mtekman can you please bump the version number.

**Comment by bgruening** [comment]
error seems to be: `|  U.error("No local minima was accepted. Recommend checking the plot "`

**Comment by mtekman** [comment]
@bgruening  - the wrapper version for the extract tool was bumped because it adds extra functionality (accepts paired inputs), but the whitelist tool has not changed because it was a fix rather than a feature. (i.e. the last release of whitelist had an error that went unnoticed but is now fixed).

Should I still bump it? I am worried that users might accidentally use the previous version of the tool

**Comment by bgruening** [comment]
I guess its on the edge, it changes reproducibility ....
Can you look at the failing test?

File: tools/umi_tools/macros.xml [hunk 1]
```diff
@@ -1,5 +1,41 @@
...
+            <when value="first_read_only"/>
+            <when value="both_reads">
+                <param name="bc_pattern2" argument="--bc-pattern2" type="text" value="" label="Barcode pattern for second read"
+                       help="Use this option to specify the format of the UMI/barcode for
+                             the second read pair if required." >
+                </param>
+            </when>
+        </conditional>
+    </macro>
+    <xml name="macro_input_types">
```
Comment (bgruening) [inline tools/umi_tools/macros.xml]: I would also not call it macro, as it is part of the macro file.

**Comment by bgruening** [comment]
Thanks @mtekman

**Pull request merged**